app/models/concerns/profile_tailoring.rb

```diff
@@ -33,4 +33,13 @@ def update_os_minor_version(version)
 
     update!(os_minor_version: version)
   end
+
+  def tailoring_valid?
+    rules_and_groups = rules + RuleGroup.where(id: rules.includes(:rule_group).flat_map(&:rule_group_ancestor_ids))
```

Comment:
Thinking about converting this to a union, this way the `rules_and_groups` would be an SQL statement and not resolved to an array. Just not sure about the `rule_group_ancestor_ids`.

Why is the `flat_map` needed?

Reply:
Can you union Rule objects with RuleGroup objects using SQL? I thought that an array was the only way to combine the Rules and RuleGroups together since they're not the same type of object 

The `flat_map` is needed because `rule_group_ancestor_ids` returns an array of RuleGroup ids and it's being called for every `rule` in `rules` so if we did `rules.includes(:rule_group).map(&:rule_group_ancestor_ids)` instead of `flat_map`, it'd look something like this `[[id1, id2], [id3, id4, id5], [id6]] `